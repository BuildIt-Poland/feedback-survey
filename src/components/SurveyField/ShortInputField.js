import React from 'react';
import { FormField } from 'grommet';

import { fontWeightNormal, colorWhite } from '../../styles/variables';
import { formField } from '../../types';

export const inputTheme = {
  global: {
    input: {
      weight: fontWeightNormal
    },
    focus: {
      border: {
        color: colorWhite
      }
    }
  }
};

const ShortInputField = ({ field }) => {
  return (
    <FormField name={field.id} label={field.content} required={field.required} placeholder="Type your answer here" />
  );
};

ShortInputField.propTypes = {
  field: formField.isRequired
};

export default ShortInputField;
