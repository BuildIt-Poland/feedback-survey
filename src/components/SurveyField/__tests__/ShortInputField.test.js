import React from 'react';
import { create } from 'react-test-renderer';

import ShortInputField from '../ShortInputField';

describe('COMPONENT - SurveyField ShortInputField', () => {
  it('renders correctly', () => {
    const field = {
      id: '123',
      type: '',
      content: 'Question content',
      required: false
    };
    const component = create(<ShortInputField field={field} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
