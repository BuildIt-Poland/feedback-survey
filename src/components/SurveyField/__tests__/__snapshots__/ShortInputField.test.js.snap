// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SurveyField ShortInputField renders correctly 1`] = `
<div
  className="sc-bdVaJa fIyeZW FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
>
  <label
    className="StyledText-sc-1sadyjn-0 bMXxti"
  >
    Question content
  </label>
  <div
    className="StyledBox-sc-13pk1d4-0 gZSidW"
  >
    <div
      className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
    >
      <input
        autoComplete="off"
        className="StyledTextInput-sc-1x30a0s-0 bYBgFP"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
      />
    </div>
  </div>
</div>
`;
